test/api/tracker/api_test.go

```diff
@@ -0,0 +1,192 @@
+				// if API service then print Not connected to Jira and pass the test
+				// else fail the test
+				if r.Connected == false {
+					t.Logf("Not connected to Jira(Thus passing the API Test)")
+				}
+				if r.Connected == true {
+					t.Errorf(err.Error())
+				}
+			}
+
+			for _, projects := range projectsList {
+
+				// convert project Id's to uint.
+				projectID, err := strconv.Atoi(projects.ID)
+				if err != nil {
+					t.Errorf(err.Error())
+				}
+
+				_, err = Tracker.GetProjects(r.ID, uint(projectID))
+				if err != nil {
+					if r.Connected == false {
+						t.Logf("Not connected to Jira(Thus passing the API Test)")
+					}
+					if r.Connected == true {
+						t.Errorf(err.Error())
+					}
+				}
+
+				_, err = Tracker.ListProjectIssueTypes(r.ID, uint(projectID))
+				if err != nil {
+					if r.Connected == false {
+						t.Logf("Not connected to Jira(Thus passing the API Test)")
+					}
+					if r.Connected == true {
+						t.Errorf(err.Error())
+					}
+				}
+			}
+
+			// Delete identity and tracker.
+			assert.Must(t, Tracker.Delete(r.ID))
+			assert.Must(t, Identity.Delete(identity.ID))
+
+			// Check if the Tracker is present even after deletion or not.
+			_, err = Tracker.Get(r.ID)
+			if err == nil {
+				t.Errorf("Resource exits, but should be deleted: %v", r)
+			}
+		})
+	}
+}
+
+func TestTrackerList(t *testing.T) {
+	createdTrackers := []api.Tracker{}
+
+	for _, r := range Samples {
+		identity := api.Identity{
+			Kind: r.Kind,
+			Name: r.Identity.Name,
+		}
+		assert.Must(t, Identity.Create(&identity))
+		r.Identity.ID = identity.ID
+
+		assert.Must(t, Tracker.Create(&r))
+		createdTrackers = append(createdTrackers, r)
+	}
+
+	// List Trackers.
+	got, err := Tracker.List()
+	if err != nil {
+		t.Errorf(err.Error())
+	}
+
+	// check if created Trackers are in the list we got from Tracker.List()
+	for _, createdTracker := range createdTrackers {
+		found := false
+		for _, retrievedTracker := range got {
+			if assert.FlatEqual(createdTracker.ID, retrievedTracker.ID) {
+				found = true
+				break
+			}
+		}
+		if !found {
+			t.Errorf("Expected tracker not found in the list: %v", createdTracker)
+		}
+	}
+
+	// Delete related trackers and identities.
+	for _, tracker := range createdTrackers {
+		assert.Must(t, Tracker.Delete(tracker.ID))
+		assert.Must(t, Identity.Delete(tracker.ID))
+	}
+}
+
+func AssertEqualTrackers(t *testing.T, got *api.Tracker, expected api.Tracker) {
+	if got.Name != expected.Name {
+		t.Errorf("Different Tracker Name Got %v, expected %v", got.Name, expected.Name)
+	}
+	if got.URL != expected.URL {
+		t.Errorf("Different Url Got %v, expected %v", got.URL, expected.URL)
+	}
+	if got.Kind != expected.Kind {
+		t.Errorf("Different Kind Got %v, expected %v", got.Kind, expected.Kind)
+	}
+	if got.Connected != expected.Connected {
+		t.Errorf("Different Connected Got %v, expected %v", got.Connected, expected.Connected)
+	}
+	if got.LastUpdated != expected.LastUpdated {
```

Comment:
Just a note on this field, I'd prefer not include LastUpdated field in Equality check (it is similar to e.g. CreatedAt which we don't assert neither).

Reply:
Will update